contracts/cw-core/src/state.rs

```diff
@@ -49,8 +67,15 @@ pub const PAUSED: Item<Expiration> = Item::new("paused");
 
 /// The voting module associated with this contract.
 pub const VOTING_MODULE: Item<Addr> = Item::new("voting_module");
-/// The proposal modules assocaited with this contract.
-pub const PROPOSAL_MODULES: Map<Addr, Empty> = Map::new("proposal_modules");
+
+/// The proposal modules associated with this contract.
+pub const PROPOSAL_MODULES: Map<Addr, ProposalModule> = Map::new("proposal_modules_v1");
+
+/// The count of active proposal modules associated with this contract.
+pub const ACTIVE_PROPOSAL_MODULE_COUNT: Item<u64> = Item::new("active_proposal_module_count");
+
+/// The count of total proposal modules associated with this contract.
+pub const TOTAL_PROPOSAL_MODULE_COUNT: Item<u64> = Item::new("total_proposal_module_count");
```

Comment:
We can use `u32` here safely. The WASM VM is 32 bit so using a u64 here is going to generate some weird machine code / doesn't actually improve the safety.